UM-USElab/gradecraft-development
# Ability to copy a course with learning objectives

### Status
**IN DEVELOPMENT**

### Description
* Copying a course with learning objectives as an admin should result in the copied course having learning objectives from the original course with any linked assignments copied over as well.
* Currently, the learning objectives model does not have a `copy` helper for ModelCopier to perform the copying. 
* Now, a `copy` helper has been added to the learning objectives model and linked models
* Note:
  * This implementation does not account for the associations of students and grades in a course with learning objectives (i.e. learning objective observed outcomes, learning objective cumulative outcomes) and will not copy these associations in a course correctly. Due to this the "Copy + Students" option will not be displayed for a course with learning objectives in the options dropdown, in the overview page for courses (i.e. /courses/overview).

### Migrations
NO

### Steps to Test or Reproduce
1. As an admin copy a course with learning objectives.
2. The resulting copy should have the learning objectives from the original course along with any linked assignments in the original course.

### Impacted Areas in Application
* Learning Objective model (models/learning_objective.rb) and associated models 

======================
Closes #4326
